qt/scientific_interfaces/Inelastic/Analysis/IDAFunctionParameterEstimation.cpp

```diff
@@ -57,30 +57,40 @@ void IDAFunctionParameterEstimation::addParameterEstimationFunction(std::string
                                                                     ParameterEstimateSetter function) {
   m_funcMap.insert(std::make_pair(functionName, std::move(function)));
 }
-// Estimate the function parameters for the input function
-// If the input function exists in the stored map it will update the function
-// parameters in-place.
-void IDAFunctionParameterEstimation::estimateFunctionParameters(Mantid::API::IFunction_sptr &function,
-                                                                const DataForParameterEstimation &estimationData) {
+
+void IDAFunctionParameterEstimation::estimateFunctionParameters(
+    Mantid::API::IFunction_sptr const &function, const DataForParameterEstimationCollection &estimationData) {
   if (!function) {
     return;
   }
 
+  // Estimate function parameters - parameters are updated in-place.
   if (auto composite = std::dynamic_pointer_cast<Mantid::API::CompositeFunction>(function)) {
+    if (estimationData.size() != composite->nFunctions()) {
+      return;
+    }
+
     for (auto i = 0u; i < composite->nFunctions(); ++i) {
       auto childFunction = composite->getFunction(i);
-      estimateSingleFunctionParameters(childFunction, estimationData, composite, i);
+      estimateFunctionParameters(childFunction, estimationData[i]);
     }
-  } else {
-    estimateSingleFunctionParameters(function, estimationData);
   }
 }
 
-void IDAFunctionParameterEstimation::estimateSingleFunctionParameters(
-    Mantid::API::IFunction_sptr &function, const DataForParameterEstimation &estimationData,
-    std::optional<Mantid::API::CompositeFunction_sptr> composite, std::optional<std::size_t> functionIndex) {
-  if (function) {
-    auto const parameterEstimatorName = nameForParameterEstimator(function, composite, functionIndex);
+void IDAFunctionParameterEstimation::estimateFunctionParameters(
+    Mantid::API::IFunction_sptr const &function, const DataForParameterEstimation &estimationData,
+    std::optional<Mantid::API::CompositeFunction_sptr> parentComposite, std::optional<std::size_t> functionIndex) {
```

Comment:
It doesn't look from a quick grep that these optional arguments are ever used - is that correct?

Reply:
They are sometimes used from inside itself:

https://github.com/mantidproject/mantid/blob/2202b8e144192f4aa799d0ceb5dcce212e031cf7/qt/scientific_interfaces/Inelastic/Analysis/IDAFunctionParameterEstimation.cpp#L90